test/gauge-proxy.ts

```diff
@@ -0,0 +1,88 @@
+/* eslint-disable no-undef */
+import { Signer, Contract } from "ethers"
+const { ethers, network } = require("hardhat");
+const hre = "hardhat";
+import { BigNumber } from "@ethersproject/bignumber";
+import chai from "chai";
+import { expect } from "chai";
+import { log } from "./utils/log";
+import {setupSigners} from "./utils/static";
+import {getContractAt, connectToGaugeProxyV2, connectToGaugeProxyV3} from "./mocks/Gauge"; 
+
+const doGaugeTest = () => {
+
+    let timelockSigner: Signer
+    let governanceSigner: Signer
+    let strategistSigner: Signer
+
+    let timelock_addr: string
+    let governance_addr: string
+    let strategist_addr: string
+    let snowball_addr = "0xC38f41A296A4493Ff429F1238e030924A1542e50"; 
+    let iceQueen_addr = "0xB12531a2d758c7a8BF09f44FC88E646E1BF9D375"; 
+    let gaugeproxyv2_addr = "0x215D5eDEb6A6a3f84AE9d72962FEaCCdF815BF27";
+    let SnowconeToken_addr: string; 
+
+    let Snowball: Contract; 
+    let IceQueen: Contract;
+    let GaugeProxyV2: Contract;
+    let GaugeProxyV3: Contract;
+
+    let numIceQueenPools: number;
+
+    describe("Deploys the Gauge that takes multiple rewards:", async () => {
+        before(async () => {
+
+            [timelockSigner, governanceSigner, strategistSigner] = await setupSigners(); 
+            timelock_addr = await timelockSigner.getAddress();
+            governance_addr = await governanceSigner.getAddress();
+
+            [Snowball, IceQueen] = await getContractAt(snowball_addr, iceQueen_addr, governanceSigner); 
+            log(`\tthe number of pre-existing pools from IceQueen`);
+            numIceQueenPools = await IceQueen.poolLength() - 1;
+
+            GaugeProxyV2 = await connectToGaugeProxyV2(gaugeproxyv2_addr, governanceSigner); 
+           
+            GaugeProxyV3 = await connectToGaugeProxyV3(governance_addr); 
+
+            log(`-- Adding Snowcones to IceQueen --`);
+            SnowconeToken_addr = await GaugeProxyV3.TOKEN();
+            log(`token addr ${SnowconeToken_addr}`);
+            await IceQueen.connect(strategistSigner).add(5000000, SnowconeToken_addr, false);
```

Comment:
fails here @bayological

Reply:
`SnowconeToke_addr` is uninitialized and the recent move to its own function does not use the returned `TOKEN`